docs/design/link-sharing-service.md

```diff
@@ -0,0 +1,93 @@
+# Link Sharing Service
+
+## Abstract
+
+This design doc outlines a link sharing service that can be used by clients
+to share content with others via a simple URL.
+
+## Design
+
+### Transport
+
+The link sharing service supports HTTPS or HTTP. HTTP is intended for
+development convenience only or when offloading TLS termination to a proxy
+running on the *same host*. The link sharing service should *NOT* be deployed
+behind an off-the-box TLS terminator unless the link between the terminator and
+the link sharing service is adequately protected. See
+[Security Considerations](#security-considerations) for details.
+
+### Requests
+
+The file hosting service handles requests of the following form:
+
+`GET /<share-blob>/path/within/bucket`
+
+The `share-blob` is url-safe base64 encoding of a `Share` protobuf, which
+is defined as follows:
+
+```
+message EncryptionAccess {
+	bytes key = 1;
+
+	bytes encrypted_path_prefix = 2;
+}
+
+message Share {
+	string satellite_url = 1;
+
+	string api_key = 2;
+
+	string bucket = 3;
+
+	EncryptionAccess encryption_access = 4;
+}
+```
+
+Between the share and the path, the link sharing service has all the
+information it needs to stream data via uplink.
+
+### Caching
+
+The link sharing service does not attempt to cache data, metadata, etc. 
+
+Presumably, with a small change to the uplink library, pieces downloaded from
+storage node operators could presumably be cached locally on disk or via some
+sort of shared cache (in the presence of horizontally scaled link sharing
+services). At this time such caching seems premature and potentially 
+harmful in that:
+
+- It complicates the code.
+- It complicates deployment.
+- It decreases payout potential for storage node operators, reducing incentive.
+- The whitepaper already describes a mechanism to deal with hot objects.
```

Comment:
I would the section where that's mentioned.

Reply:
I've updated the PR with a link to the whitepaper and a mention of the specific section.